### PR TITLE
Add no-op metrics factory

### PR DESCRIPTION
**Pull request by nicu-da**
Will allow us to clean-up the code in places where metrics are not important/we want to run without metrics

### CODE REVIEW

File: observability/metrics/src/main/scala/com/daml/metrics/api/noop/Metrics.scala [hunk 1]
```diff
@@ -6,8 +6,8 @@ package com.daml.metrics.api.noop
 import java.time.Duration
 import java.util.concurrent.TimeUnit
 
-import com.daml.metrics.api.MetricHandle.Timer
 import com.daml.metrics.api.MetricHandle.Timer.TimerHandle
+import com.daml.metrics.api.MetricHandle.{Counter, Gauge, Histogram, Meter, Timer}
 import com.daml.metrics.api.MetricsContext
 
 sealed case class NoOpTimer(name: String) extends Timer {
```
Comment (lucb-da) [inline observability/metrics/src/main/scala/com/daml/metrics/api/noop/Metrics.scala:13]: Any reason to have `sealed` on this case class ?

**Review by lucb-da — APPROVED** [review]
Comment on existing code, otherwise LGTM.

File: observability/metrics/src/main/scala/com/daml/metrics/api/noop/Metrics.scala (hunk 1 above)
Comment (nicu-da) [reply observability/metrics/src/main/scala/com/daml/metrics/api/noop/Metrics.scala:13]: Indeed, no real benefit.

Comment (nicu-da) [reply observability/metrics/src/main/scala/com/daml/metrics/api/noop/Metrics.scala:13]: Will remove it

**Pull request merged**